cpp/src/io/utilities/column_utils.cuh

```diff
@@ -67,13 +67,17 @@ rmm::device_uvector<column_device_view> create_leaf_column_device_views(
       size_type index) mutable {
       col_desc[index].parent_column = parent_col_view.begin() + index;
       column_device_view col        = parent_col_view.column(index);
-      if (col_desc[index].stats_dtype != dtype_byte_array) {
-        // traverse till leaf column
-        while (col.type().id() == type_id::LIST or col.type().id() == type_id::STRUCT) {
-          col = (col.type().id() == type_id::LIST)
-                  ? col.child(lists_column_view::child_column_index)
-                  : col.child(0);
+      // traverse till leaf column
+      while (col.type().id() == type_id::LIST or col.type().id() == type_id::STRUCT) {
+        auto child = (col.type().id() == type_id::LIST)
```

Comment:
Can we sprinkle more `const` throughout here? I kept expecting to see a mutation for several variables because they were non-const.